2025/using-components-2/demo/3-typescript/src/vite-env.d.ts

```diff
@@ -0,0 +1,4 @@
+/// <reference types="vite/client" />
+/// <reference types="@arcgis/core/interfaces" />
```

Comment:
I'm not 100% sure if this is necessary but I could be wrong https://developers.arcgis.com/javascript/latest/typescript-setup/#first-typescript-file

> The API's types are automatically inferred, you don't need to set type annotations for every declaration.